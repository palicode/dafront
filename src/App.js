
import React, { Component } from 'react';
import { HashRouter, Route, Switch } from 'react-router-dom';
import {
  LayoutManagerWithViewController,
  NavigationProvider,
  withNavigationViewController,
} from '@atlaskit/navigation-next';
import PropTypes from 'prop-types'

import ProjectMenu, { projectHomeView } from './navigation/ProjectMenu'
import ProductMenu, { productHomeView } from './navigation/ProductMenu'
import ConfigSubMenu, { configView } from './navigation/ConfigSubMenu'
import GlobalNavigation from './navigation/GlobalMenu'

class App extends Component {
  componentDidMount() {
    const { navigationViewController } = this.props;
    navigationViewController.addView(productHomeView);
    navigationViewController.addView(configView);
    navigationViewController.addView(projectHomeView);
    navigationViewController.setInitialPeekViewId(productHomeView.id);
  }

  render() {
    return (
      <LayoutManagerWithViewController globalNavigation={GlobalNavigation}>
        <Switch>
          <Route path="/projects/my-project" component={ProjectMenu} />
          <Route path="/issues" component={ConfigSubMenu} />
          <Route path="/" component={ProductMenu} />
        </Switch>
      </LayoutManagerWithViewController>
    );
  }
}

App.propTypes = {
  navigationViewController: PropTypes.func.isRequired,
}

const AppWithNavigationViewController = withNavigationViewController(App);

export default () => (
  <HashRouter>
    <NavigationProvider>
      <AppWithNavigationViewController />
    </NavigationProvider>
  </HashRouter>
);
