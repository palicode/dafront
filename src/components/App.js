import React from 'react';

import '../styles/App.css';

const App = () => (
  <div>
    <h1>My React live!</h1>
  </div>
);

export default App;
