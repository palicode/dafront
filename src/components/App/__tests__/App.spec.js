import React from 'react';
import { render } from 'react-testing-library';
import App from '../index';

describe('App', () => {
  it('should be mounted with no props', () => {
    const wrapper = render(<App />);
    expect()
    expect(wrapper).toMatchSnapshot();
  })
})
