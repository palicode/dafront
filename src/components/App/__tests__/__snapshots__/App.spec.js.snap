// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should be mounted with no props 1`] = `
Object {
  "baseElement": <body>
    <div>
      <div>
        <h1>
          This is Nucleotid.
        </h1>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <h1>
        This is Nucleotid.
      </h1>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
